api/app/v1/lineage/LineageController.scala

```diff
@@ -100,7 +103,11 @@ class LineageController @Inject()(cc: LineageControllerComponents)
         // Produce a proper HTTP response for different success cases from LineageService
         lineage.transform {
           // A failure fetching lineage produces a 500 error code
-          case Failure(_) => Success(InternalServerError(s"""Unexpected error constructing lineage for table "$name""""))
```

Comment:
We don't need both Failure(_) and Failure(ex) . Failure is always raised due to an exception. However _ covers exception and anything else but if we just catch on Failure(_) we loose reference to exception. 
So Catching on Failure(ex) is sufficient. Please remove Failure(_). 
Also  if we keep both of  these catches , errors will be handled by first case one (since its more broader)